src/orion/executor/single_backend.py

```diff
@@ -5,8 +5,49 @@
 
 """
 import functools
+import traceback
 
-from orion.executor.base import BaseExecutor
+from orion.executor.base import (
+    AsyncException,
+    AsyncResult,
+    BaseExecutor,
+    ExecutorClosed,
+    Future,
+)
+
+
+class _Future(Future):
```

Comment:
well yes, single_backends does not have a true future

Reply:
It's not really asynchronous indeed, but the different methods should be tested anyhow. There is a common interface across the Futures, like successful raising ValueError if the future is not ready. But in this case, it's never _not ready_, right? :thinking:

Reply:
yeah, I mean given that multprocess is builtin in python do we really need the single_backend ?
We could simply default to the threaded implementation.

I understood that single_backend was more for testing that for actual use

Reply:
Single process can be helpful to debug user's functions in case multiprocess/multithread causes issues with the stack trace.

Reply:
single_backend should work almost identically to the others now.
There is a single exception when wait(0.001) is called, the single_backend will always have a result (i.e timeout is kind of irrelevant since we have no way of stopping execution once we start)

The difference is captured in the tests so that should  be fine

Reply:
Sounds good! Thanks a lot!